docs/versioned_docs/version-zenith/quickstart/822194-daggerize.mdx

```diff
@@ -0,0 +1,234 @@
+    cd greetings-api
+    ```
+
+2. Initialize a Dagger module at the root of the repository:
+
+    ```shell
+    dagger init
+    ```
+
+    You should see a new file at the root of your repository, named `dagger.json`. View its contents:
+
+    ```shell
+    cat dagger.json
+    ```
+
+    You should see the following:
+
+    ```json
+    {
+      "name": "greetings-api",
+      "engineVersion": "v0.9.11"
+    }
+    ```
+
+    Finally, commit the file to your repository:
+
+    ```shell
+    git add dagger.json
+    git commit -m 'dagger init'
+    ```
+
+    Congratulations! Your repository is now a Dagger Module.
+
+3. Now that your project is a Dagger Module, you can install other modules into it, as dependencies. A dependency is just a Dagger Module installed into another. As an example, install the `hello` module from earlier:
+
+    ```shell
+    dagger install github.com/shykes/daggerverse/hello
+    ```
+
+    Once the command completes, look at the `dagger.json` file again. You should see that a new dependency has been added, similar to that shown below:
+
+    ```json
+    {
+      "name": "greetings-api",
+      "dependencies": [
+        {
+          "name": "hello",
+          "source": "github.com/shykes/daggerverse/hello@7884cbc1941f85a7c714fc7f1bcc5c148842c21e"
+        }
+      ],
+      "engineVersion": "v0.9.11"
+    }
+    ```
+
+    :::note
+    The exact Git commit for the module version is recorded in `dagger.json`. Dagger enforces "version pinning", which guarantees that the module version you install is the one you'll be using.
+    :::
+
+    Go ahead and commit this change as well:
+
+    ```shell
+    git add dagger.json
+    git commit -m 'dagger install github.com/shykes/daggerverse/hello'
+    ```
+
+4. Now that your module has its first dependency installed, you can start using it! To do so, simply refer to it by its short name - in this case, `hello`.
+
+    List available functions:
+
+    ```shell
+    dagger -m hello functions
+    ```
+
+    You should see a familiar result:
+
+    ```shell
+    Name    Description
+    hello   Say hello to the world!
+    ```
+
+    Call a function:
+
+    ```shell
+    dagger -m hello call hello
+    ```
+
+    Again, the output should be familiar:
+
+    ```shell
+    hello, world!
+    ```
+
+5. The `hello` module is a good example to start with, but it doesn't have any functions to work with a Go project. To Daggerize the Go build for this example project, install the Go builder module from a previous section as another dependency:
+
+    ```shell
+    dagger install github.com/kpenfound/dagger-modules/golang
+    ```
+
+    Then, call the `Build()` function from the Go builder module to build the project and chain other function calls to export the built binary to the host:
+
+    ```shell
+    dagger -m golang --proj . call \
+      # call the go builder function
+      build --os=$(uname -s) --arch=$(uname -m) --args . \
```

Comment:
@shykes @kpenfound just a note that this didn't work on my host, error was `go: unsupported GOOS/GOARCH pair Linux/x86_64`

Reply:
For this quickstart, should I hardcode these to `linux` and `amd64` instead of using `uname`?

Reply:
You should be able to get rid of the `--arch` flag. It'll default correctly. If the formatting of the os from `uname -s` doesn't work, I can fix that in the module